File: src/main/kotlin/naver/finantial/p2/java/Solution.java
package naver.finantial.p2.java;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int[] solution(String s, String[] times) {
        final String[] date = s.split(":");
        final LocalDateTime dateTime = LocalDateTime.of(
                Integer.parseInt(date[0]),
                Integer.parseInt(date[1]),
                Integer.parseInt(date[2]),
                Integer.parseInt(date[3]),
                Integer.parseInt(date[4]),
                Integer.parseInt(date[5])
        );

        final List<LocalDateTime> dateTimes = new ArrayList<>();
        dateTimes.add(dateTime);

        for (String time : times) {
            final String[] split = time.split(":");
            LocalDateTime last = dateTimes.get(dateTimes.size() - 1);
            last = last.plusSeconds(Integer.parseInt(split[3]));
            last = last.plusMinutes(Integer.parseInt(split[2]));
            last = last.plusHours(Integer.parseInt(split[1]));
            last = last.plusDays(Integer.parseInt(split[0]));
            dateTimes.add(last);
        }

        final int period = dateTimes.get(dateTimes.size() - 1).getDayOfMonth() - dateTimes.get(0).getDayOfMonth() + 1;
        int[] answer = {1, period};
        for (int i = 1; i < dateTimes.size(); i++) {
            if (dateTimes.get(i).getDayOfMonth() - dateTimes.get(i - 1).getDayOfMonth() > 1) {
                answer[0] = 0;
                break;
            }
        }

        return answer;
    }
}

File: src/main/kotlin/naver/finantial/p2/Solution.kt
package naver.finantial.p2

import java.time.LocalDateTime

fun main() {
    println(Solution().solution("2021:04:12:16:08:35", arrayOf("01:06:30:00", "01:04:12:00")).toMutableList())
}

class Solution {

    fun solution(s: String, times: Array<String>): IntArray {

        val date = s.split(":")
        val dateTime = LocalDateTime.of(
            date[0].toInt(),
            date[1].toInt(),
            date[2].toInt(),
            date[3].toInt(),
            date[4].toInt(),
            date[5].toInt(),
        )

        var dateTimes: MutableList<LocalDateTime> = mutableListOf()
        dateTimes.add(dateTime)

        for (time in times) {
            val split = time.split(":")
            var last = dateTimes.last()
            println("+++")
            println(split)
            println(last)
            last = last.plusSeconds(split[3].toLong())
            last = last.plusMinutes(split[2].toLong())
            last = last.plusHours(split[1].toLong())
            last = last.plusDays(split[0].toLong())
            println(last)
            dateTimes.add(last)
        }

        println(dateTimes)

        val period = dateTimes.last().dayOfMonth - dateTimes.first().dayOfMonth + 1
        var answer: MutableList<Int> = mutableListOf(1, period)
        for (i in 1 until dateTimes.size) {
            if (dateTimes[i].dayOfMonth - dateTimes[i - 1].dayOfMonth > 1) {
                answer[0] = 0
                break
            }
        }

        return answer.toIntArray()
    }
}
